connector/elasticapmconnector/config.go

```diff
@@ -0,0 +1,282 @@
+	lsmconfig "github.com/elastic/opentelemetry-collector-components/processor/lsmintervalprocessor/config"
+	"go.opentelemetry.io/collector/component"
+)
+
+var _ component.Config = (*Config)(nil)
+
+type Config struct {
+	// Aggregation holds configuration related to aggregation of Elastic APM
+	// metrics from other signals.
+	Aggregation *AggregationConfig `mapstructure:"aggregation"`
+}
+
+type AggregationConfig struct {
+	// Directory holds a path to the directory that is used for persisting
+	// aggregation state to disk. If Directory is empty, in-memory storage
+	// is used.
+	Directory string `mapstructure:"directory"`
+
+	// MetadataKeys holds a list of client.Metadata keys that will be
+	// propagated through to aggregated metrics. Only the listed metadata
+	// keys will be propagated.
+	//
+	// Entries are case-insensitive, and duplicated entries will trigger
+	// a validation error.
+	MetadataKeys []string `mapstructure:"metadata_keys"`
+}
+
+func (cfg Config) Validate() error {
+	lsmConfig := cfg.lsmConfig()
+	return lsmConfig.Validate()
+}
+
+func (cfg Config) lsmConfig() *lsmconfig.Config {
+	lsmConfig := &lsmconfig.Config{
+		Intervals: []lsmconfig.IntervalConfig{{
+			Duration: time.Minute,
+			Statements: []string{
+				// TODO(axw) confirm that it's ok to change metricset.interval
+				// from a resource attribute to datapoint attribute. Resource is
+				// meant to be about the producer of the signal, I don't think
+				// this fits.
+				`set(attributes["metricset.interval"], "1m")`,
+				`set(attributes["data_stream.dataset"], Concat([attributes["metricset.name"], "1m"], "."))`,
+				`set(attributes["processor.event"], "metric")`,
+			},
+		}, {
+			Duration: 10 * time.Minute,
+			Statements: []string{
+				`set(attributes["metricset.interval"], "10m")`,
+				`set(attributes["data_stream.dataset"], Concat([attributes["metricset.name"], "10m"], "."))`,
+				`set(attributes["processor.event"], "metric")`,
+			},
+		}, {
+			Duration: 60 * time.Minute,
+			Statements: []string{
+				`set(attributes["metricset.interval"], "60m")`,
+				`set(attributes["data_stream.dataset"], Concat([attributes["metricset.name"], "60m"], "."))`,
+				`set(attributes["processor.event"], "metric")`,
+			},
+		}},
+	}
+	if cfg.Aggregation != nil {
+		lsmConfig.Directory = cfg.Aggregation.Directory
+		lsmConfig.MetadataKeys = cfg.Aggregation.MetadataKeys
+	}
+	return lsmConfig
+}
+
+func (cfg Config) signaltometricsConfig() *signaltometricsconfig.Config {
+	// serviceResourceAttributes is the resource attributes included in
+	// service-level aggregated metrics.
+	serviceResourceAttributes := []signaltometricsconfig.Attribute{
+		{Key: "service.name"},
+		{Key: "deployment.environment"}, // service.environment
+		{Key: "telemetry.sdk.language"}, // service.language.name
+		{Key: "agent.name"},             // set via elastictraceprocessor
```

Comment:
We should set a default value like `unknown` as logs and metrics don't go through the `elastictraceprocessor`. See also the discussion we had in the context of the EDOT k8s gateway mode: https://github.com/elastic/elastic-agent/pull/6444#discussion_r1925699196.

I'm so glad that we can now centralize the configuration to make sure it's consistent everywhere.

Reply:
Hmmm so I will set it, but I think just defaulting to `unknown` will be a problem too: that means metrics and traces coming from the same client will result in different aggregated service metrics. We might need to move the resource attribute enrichment logic out of `elastictrace` to a separate processor.

Reply:
Alternatively (preferably) we update the UI to stop expecting `agent.name`, and have it work with either `agent.name` or the `telemetry.sdk.*` SemConv attributes instead.